src/tbp/monty/frameworks/environments/embodied_data.py

```diff
@@ -554,160 +554,59 @@ def get_good_view(
-            )
-            # Continue moving to a close distance to the object
-            while not close_enough:
-                logging.debug("moving closer!")
-                self._observation, proprio_state = self.dataset[action]
-                self.motor_system._state = (
-                    MotorSystemState(proprio_state) if proprio_state else None
-                )
-                action, close_enough = self.motor_system._policy.move_close_enough(
-                    self._observation,
-                    sensor_id,
-                    target_semantic_id=self.primary_target["semantic_id"],
-                    multiple_objects_present=multiple_objects_present,
-                )
-
-        on_target_object = self.motor_system._policy.is_on_target_object(
-            self._observation,
-            sensor_id,
+        positioning_procedure = GetGoodView(
+            agent_id=self.motor_system._policy.agent_id,
+            desired_object_distance=self.motor_system._policy.desired_object_distance,
+            good_view_percentage=self.motor_system._policy.good_view_percentage,
+            multiple_objects_present=self.num_distractors > 0,
+            sensor_id=sensor_id,
             target_semantic_id=self.primary_target["semantic_id"],
-            multiple_objects_present=multiple_objects_present,
+            allow_translation=allow_translation,
+            max_orientation_attempts=max_orientation_attempts,
+            # TODO: Remaining arguments are unused but required by BasePolicy.
+            #       These will be removed when PositioningProcedure is split from
+            #       BasePolicy
+            #
+            # Note that if we use rng=self.rng below, then the following test will
+            # fail:
+            #   tests/unit/evidence_lm_test.py::EvidenceLMTest::test_two_lm_heterarchy_experiment  # noqa: E501
+            # The test result seems to be coupled to the random seed and the
+            # specific sequence of rng calls (rng is called once on GetGoodView
+            # initialization).
+            rng=np.random.RandomState(),
+            action_sampler_args=dict(actions=[LookUp]),
+            action_sampler_class=UniformlyDistributedSampler,
+            switch_frequency=0.0,
         )
-        num_attempts = 0
-        while not on_target_object and num_attempts < max_orientation_attempts:
-            actions = self.motor_system._policy.orient_to_object(
-                self._observation,
-                sensor_id,
-                target_semantic_id=self.primary_target["semantic_id"],
-                multiple_objects_present=multiple_objects_present,
-                state=self.motor_system._state,
-            )
-            for action in actions:
+        result = positioning_procedure.positioning_call(
+            self._observation, self.motor_system._state
+        )
+        while not result.terminated and not result.truncated:
```

Comment:
I don't remember if you responded to my previous comment about this, so if so, you can ignore this.

I feel like this reads better if you factor out the `not` using [De Morgan's law][de-morgan].
```suggestion
        while not (result.terminated or result.truncated):
```
It could just be my brain though.

[de-morgan]: https://en.wikipedia.org/wiki/De_Morgan%27s_laws

Reply:
I did respond. I'm waiting for a third vote to break the tie between our two opinions on this 🙂.

Reply:
If no one else has an option, then I'd say author's choice.

Reply:
I didn't comment since I am happy with either one. To me they are both equally easy to read.